### ISSUE TITLE
C tokenizer exited with non-empty token stack

### ISSUE THREAD

**Issue by halfak** [earwig/mwparserfromhell#164]
Error while processing [2013 AIBA World Boxing Championships – Light welterweight](https://en.wikipedia.org/wiki/2013_AIBA_World_Boxing_Championships_%E2%80%93_Light_welterweight) @ [726997736](https://en.wikipedia.org/wiki/?oldid=726997736): 

```
ParserError: This is a bug and should be reported. Info: C tokenizer exited with non-empty token stack.
Traceback (most recent call last):
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/revscoring/dependencies/functions.py", line 244, in _solve
    value = dependent(*args)
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/revscoring/dependencies/dependent.py", line 52, in __call__
    return self.process(*args, **kwargs)
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/revscoring/features/wikitext/datasources/parsed.py", line 210, in _process_wikicode
    return mwparserfromhell.parse(text)
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/mwparserfromhell/utils.py", line 58, in parse_anything
    return Parser().parse(value, context, skip_style_tags)
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/mwparserfromhell/parser/__init__.py", line 93, in parse
    tokens = self._tokenizer.tokenize(text, context, skip_style_tags)
mwparserfromhell.parser.ParserError: This is a bug and should be reported. Info: C tokenizer exited with non-empty token stack.
```

```
$ python
Python 3.4.3 (default, Oct 14 2015, 20:28:29) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import mwparserfromhell
>>> import mwapi
>>> mwparserfromhell.__version__
'0.4.3'
>>> text = mwapi.Session("https://en.wikipedia.org").get(action='query', prop='revisions', revids=726997736, rvprop=['content'], formatversion=2)['query']['pages'][0]['revisions'][0]['content']
Sending requests with default User-Agent.  Set 'user_agent' on mwapi.Session to quiet this message.
>>> wikicode = mwparserfromhell.parse(text)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/mwparserfromhell/utils.py", line 58, in parse_anything
    return Parser().parse(value, context, skip_style_tags)
  File "/home/halfak/env/3.4/lib/python3.4/site-packages/mwparserfromhell/parser/__init__.py", line 93, in parse
    tokens = self._tokenizer.tokenize(text, context, skip_style_tags)
mwparserfromhell.parser.ParserError: This is a bug and should be reported. Info: C tokenizer exited with non-empty token stack.
```

**Comment by lahwaacz** [comment]
Most likely a duplicate of #40, the same workaround (`skip_style_tags=True`) applies.

**Comment by earwig** [comment]
Fixed in cd4f90e663fa421b836e93ddc56e4a573eefb664, same as #147 and #158. Slipped by me the first time around (probably because it has the same title as #158 :P)
